src/app/login/login.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+  myForm: FormGroup;
+  constructor(private fb: FormBuilder, private _router: Router) { }
+
+  ngOnInit() {
+    this.myForm = this.fb.group({
+      email: ['', [Validators.required, Validators.pattern('[a-z0-9.@]*'), Validators.minLength(7)]],
```

Comment:
while it is probably fine.  I'm not sure we need a min length validator on the email